### PR TITLE
Added general method unit tests

### PR DESCRIPTION
**Pull request by aburrell**
# Description
Partially addresses #3 by adding unit tests for many of the general methods and simple file loading.

# Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality or documentation)
- This change requires a documentation update

# How Has This Been Tested?

Running the new unit tests

## Test Configuration
* Operating system: OS X Mojave
* Version number: Python 3.9
* Any details about your local setup that are relevant: requires pysat develop branch.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have linted the files updated in this pull request
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by jklenzing** [comment]
I should have caught that `prep_dir` was used elsewhere in the pysat pull.  Should have been deprecated there instead of just moved, which is why these tests are breaking.  We can either use this with the develop branch of pysat, or switch back to the old path here and add a pass-through with DeprecationWarning in pysat develop to maintain usage.

**Comment by aburrell** [comment]
It doesn't look like that's why it's failing, though.  It looks like it's failing because the tests aren't being skipped like they should be.

**Comment by aburrell** [comment]
@jklenzing it's passing now!

File: pysatMadrigal/tests/test_methods_general.py [hunk 1]
```diff
@@ -1,34 +1,265 @@
...
+        with pytest.raises(ValueError) as verr:
+            general.get_remote_filenames(**self.kwargs)
+
+        assert str(verr).find("unknown date_array supplied") >= 0
+        return
+
+
+@pytest.mark.skipif(((pysat_version_minor <= 0) & (pysat_version_bug <= 1))
+                    | (pysat_version_major < 3),
+                    reason="Test requires pysat 3.0.2+")
```
Comment (jklenzing) [inline pysatMadrigal/tests/test_methods_general.py:166]:
```suggestion
```

File: pysatMadrigal/tests/test_methods_general.py [hunk 2]
```diff
@@ -1,34 +1,265 @@
...
+class TestSimpleFiles(object):
+    """Tests for general methods with simple files."""
+
+    def setup(self):
+        """Create a clean testing setup."""
+
+        # Create testing directory
+        self.data_path = tempfile.TemporaryDirectory()
+        testing.prep_dir(self.data_path.name)
```
Comment (jklenzing) [inline pysatMadrigal/tests/test_methods_general.py:175]:
```suggestion
```

File: pysatMadrigal/tests/test_methods_general.py [hunk 3]
```diff
@@ -1,34 +1,265 @@
...
+# Full author list can be found in .zenodo.json file
+# DOI:10.5281/zenodo.3824979
+# ----------------------------------------------------------------------------
+"""Unit tests for the general instrument methods."""
+
+import gzip
+import tempfile
+import os
+from pysat import __version__ as pysat_version
+from pysat.utils import testing
```
Comment (jklenzing) [inline pysatMadrigal/tests/test_methods_general.py:12]:
```suggestion
```

File: pysatMadrigal/tests/test_methods_general.py [hunk 4]
```diff
@@ -1,34 +1,265 @@
...
 import pytest
 
 from pysatMadrigal.instruments.methods import general
 
+pysat_version_major = int(pysat_version.split(".")[0])
+pysat_version_minor = int(pysat_version.split(".")[1])
+pysat_version_bug = int(pysat_version.split(".")[2])
+
```
Comment (jklenzing) [inline pysatMadrigal/tests/test_methods_general.py:20]:
```suggestion
```

**Review by jklenzing — CHANGES_REQUESTED** [review]
`prep_dir` is only required if you're creating subdirectories within the temporary directory.  In the core code, this is done sometimes because a full instrument path is needed (though it's likely not consistently done this way).  Since we're using the main directory already created by `tempfile`, it's not needed here, which means that all the skip logic can be removed.  Verified this by commenting out line 175 and running locally.

**Review by jklenzing — APPROVED** [review]
Looks good! Minor comment on spelling.

File: pysatMadrigal/tests/test_methods_general.py [hunk 5]
```diff
@@ -1,34 +1,255 @@
...
+        self.kwargs[in_key] = in_val
+
+        with pytest.raises(ValueError) as verr:
+            general.download([], **self.kwargs)
+
+        assert str(verr).find(test_verr) >= 0
+        return
+
+    def test_get_remote_filenames_bad_date_array(self):
+        """Tesst raises ValueError for unexpected date_array input."""
```
[the remark this answers is not in the document]
Comment (aburrell) [reply pysatMadrigal/tests/test_methods_general.py:146]: SSSSS 🐍

**Pull request merged**